### PR TITLE
Update the module based on AWS changes to S3 bucket default settings

### PR DESCRIPTION
**Pull request by jsclarridge**
Updates the module to account for changes made by AWS in April 2023 to the default security settings of new S3 buckets.

https://aws.amazon.com/about-aws/whats-new/2022/12/amazon-s3-automatically-enable-block-public-access-disable-access-control-lists-buckets-april-2023/

See the proposed changes in the README `Upgrade Paths` section for more details about the module changes.

If approved, the plan is to release this as a major version update.

### CODE REVIEW

**Review by esacteksab — APPROVED** [review]

**Review by rpdelaney — CHANGES_REQUESTED** [review]

File: variables.tf [hunk 1]
```diff
@@ -147,3 +147,21 @@ variable "additional_lifecycle_rules" {
...
+  description = "Object ownership. Valid values: BucketOwnerEnforced, BucketOwnerPreferred or ObjectWriter."
+  type        = string
+  default     = "BucketOwnerEnforced"
+}
+
+variable "s3_bucket_acl" {
+  description = "Set bucket ACL per [AWS S3 Canned ACL](<https://docs.aws.amazon.com/AmazonS3/latest/dev/acl-overview.html#canned-acl>) list."
+  default     = null
+  type        = string
+}
```
[the remark this answers is not in the document]
Comment (rpdelaney) [reply variables.tf:167]: This is potentially helpful if setting a wrong value leads to surprising behavior. If it fails to plan, then we don't need validation.

File: README.md [hunk 2]
```diff
@@ -124,6 +128,38 @@ No modules.
 
 ## Upgrade Paths
 
+### Upgrading from 4.x.x to 5.x.x
```
[the remark this answers is not in the document]
Comment (jsclarridge) [reply README.md:131]: Can you provide suggested text to add?

File: README.md [hunk 3]
```diff
@@ -124,6 +128,38 @@ No modules.
...
+
+Version 5.x.x updates the module to account for changes made by AWS in April
+2023 to the default security settings of new S3 buckets.
+
+Version 5.x.x of this module adds the following resource and variables. How to
+use the new variables will depend on your use case.
+
```
Comment (rpdelaney) [inline README.md:138]:
```suggestion
use the new variables will depend on your use case.

Removed variables:
- `sse_algorithm`. If `kms_master_key_id` is not passed, the module will fall back to AES256 for the bucket encryption configuration.

```

File: README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (rpdelaney) [reply README.md:131]: I left a suggestion!

[the remark this answers is not in the document]
Comment (jsclarridge) [reply README.md]: Thanks. Added.

File: variables.tf [hunk 4]
```diff
@@ -147,3 +147,21 @@ variable "additional_lifecycle_rules" {
   type        = list(any)
   default     = []
 }
+
+variable "control_object_ownership" {
```
[the remark this answers is not in the document]
Comment (jsclarridge) [reply variables.tf:151]: Good catch. I updated main.tf to use this variable.

File: variables.tf (hunk 1 above)
[the remark this answers is not in the document]
Comment (jsclarridge) [reply variables.tf:167]: I like that you're thinking about this, but I don't think the validation is needed. Terraform fails to plan when I use a non-expected ACL, without the additional validation here.

File: README.md (hunk 3 above)
Comment (jsclarridge) [reply README.md:138]: Added. See note above.

**Review by rpdelaney — APPROVED** [review]

**Pull request merged**